swap_meet/vendor.py

```diff
@@ -1,2 +1,105 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        # if no inventory, default to empty list
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        # always add item to inventory
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        # remove item if it exists
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+
+    # part of wave 2
+    def get_by_id(self, id):
+        # find item in inventory by its unique id
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+
+    # wave 3
+    # removes my_item from this Vendor inventory and adds to friend inventory
+    # removes their_item from other Vendor inventory and adds it to this inventory
+    # method returns True
+    # If vendor inventories do not contain my_item or their_item
+    # return False
+    def swap_items(self, other_vendor, my_item, their_item):
+        self.other_vendor = other_vendor
+        self.my_item = my_item
+        self.their_item = their_item
```

Comment:
When you access self and use dot notation with a new attribute name, you are now creating a new attribute for this instance. 

Before `swap_items` executes, an instance of `Vendor` only has an attribute called `inventory` that you access with `self.inventory`.

After lines 35-37 execute, you add 3 new attributes to the instance of `Vendor`. 

Do you need an instance of `Vendor` to have these 3 new attributes? Why does `Vendor` need them to swap items? Take care to review what your code is doing and why with OOP. 

I see that you set these three attributes and but the method does nothing with them. Notice that there are no other occurrences of `self.other_vendor`, `self.my_item`, or `self.their_item` in the rest of the method. 

```suggestion
```